cmd/apprepository-controller/main.go

```diff
@@ -17,42 +17,108 @@ limitations under the License.
+	RepoSyncImagePullSecrets []string
+	ImagePullSecretsRefs     []corev1.LocalObjectReference
+	RepoSyncCommand          string
+	KubeappsNamespace        string
+	DBURL                    string
+	DBUser                   string
+	DBName                   string
+	DBSecretName             string
+	DBSecretKey              string
+	UserAgentComment         string
+	Crontab                  string
+	ReposPerNamespace        bool
+
+	// Args are the positional (non-flag) command-line arguments.
+	Args []string
+}
+
+// parseFlags parses the command-line arguments provided to the program.
+// Typically os.Args[0] is provided as 'programName' and os.args[1:] as 'args'.
+func parseFlags(progname string, args []string) (config *Config, output string, err error) {
+	flagSet := flag.NewFlagSet(progname, flag.ContinueOnError)
+	var buf bytes.Buffer
+	flagSet.SetOutput(&buf)
+
+	var conf Config
+
+	flagSet.StringVar(&conf.Kubeconfig, "kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
+	flagSet.StringVar(&conf.MasterURL, "master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
+	flagSet.StringVar(&conf.RepoSyncImage, "repo-sync-image", "quay.io/helmpack/chart-repo:latest", "container repo/image to use in CronJobs")
+	flagSet.StringSliceVar(&conf.RepoSyncImagePullSecrets, "repo-sync-image-pullsecrets", nil, "optional reference to secrets in the same namespace to use for pulling the image used by this pod")
+	flagSet.StringVar(&conf.RepoSyncCommand, "repo-sync-cmd", "/chart-repo", "command used to sync/delete repos for repo-sync-image")
+	flagSet.StringVar(&conf.KubeappsNamespace, "namespace", "kubeapps", "Namespace to discover AppRepository resources")
+	flagSet.BoolVar(&conf.ReposPerNamespace, "repos-per-namespace", true, "Defaults to watch for repos in all namespaces. Switch to false to watch only the configured namespace.")
+	flagSet.StringVar(&conf.DBURL, "database-url", "localhost", "Database URL")
+	flagSet.StringVar(&conf.DBUser, "database-user", "root", "Database user")
+	flagSet.StringVar(&conf.DBName, "database-name", "charts", "Database name")
+	flagSet.StringVar(&conf.DBSecretName, "database-secret-name", "kubeapps-db", "Kubernetes secret name for database credentials")
+	flagSet.StringVar(&conf.DBSecretKey, "database-secret-key", "postgresql-root-password", "Kubernetes secret key used for database credentials")
+	flagSet.StringVar(&conf.UserAgentComment, "user-agent-comment", "", "UserAgent comment used during outbound requests")
+	flagSet.StringVar(&conf.Crontab, "crontab", "*/10 * * * *", "CronTab to specify schedule")
+
+	err = flagSet.Parse(args)
+	if err != nil {
+		return nil, buf.String(), err
+	}
+	conf.Args = flagSet.Args()
+	return &conf, buf.String(), nil
+}
 
 func main() {
-	flag.Parse()
+	conf, output, err := parseFlags(os.Args[0], os.Args[1:])
+
+	if err != nil {
+		log.Fatal("got error:", err)
+		log.Fatal("output:\n", output)
```

Comment:
These look like debugging? Perhaps make it more specific to the context, something like:
```
log.Fatal("error parsing command-line arguments: ", err)
log.Fatal(output)
```